### PR TITLE
ci(gcb): add integration-daily build to GCB

### PR DESCRIPTION
**Pull request by devjgm**
Fixes: #6384

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6386)

### CODE REVIEW

File: ci/cloudbuild/builds/integration-nightly.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# ATTENTION: This build is different than most because it's designed to be run
+# as a "manual" trigger that's run on a schedule in GCB. It will be scheduled
+# to right once a day at night. For details about scheduled manual triggers see
```
Comment (devbww) [inline ci/cloudbuild/builds/integration-nightly.sh:19]: "At night" is in the eye of the beholder.

Comment (devjgm) [reply ci/cloudbuild/builds/integration-nightly.sh:19]: Just copying the existing name. Happy to rename to whatever you suggest.

Comment (coryan) [inline ci/cloudbuild/builds/integration-nightly.sh:19]: s/right once a day/run once a day/

Comment (coryan) [reply ci/cloudbuild/builds/integration-nightly.sh:19]: consider "once a day, at a time when most of the `google-cloud-cpp` developers are not working" or something like that.

Comment (devjgm) [reply ci/cloudbuild/builds/integration-nightly.sh]: SGTM. Changed all references to "nightly" to "daily".

Comment (devjgm) [reply ci/cloudbuild/builds/integration-nightly.sh]: Fixed

**Review by coryan — APPROVED** [review]

**Comment by devjgm** [comment]
Hmm. The spanner tests keep timing out. Example: https://pantheon.corp.google.com/cloud-build/builds;region=global/ef3cc3f7-4af8-4330-ac91-81f4074ca002;step=3?project=cloud-cpp-testing-resources

Will look into this and try to get one to completely successfully before merging this.

**Comment by devjgm** [comment]
I think I just need to bump the spanner test timeout, because I found a previous run of this test from Kokoro where the times were:

```
//google/cloud/spanner/integration_tests:backup_integration_test         PASSED in 1831.0s
...
//google/cloud/spanner/samples:samples                                   PASSED in 1506.8s
```

**Comment by coryan** [comment]
I think #5926 extended the integration test time for these tests, but in the GCB builds we override that:

https://github.com/googleapis/google-cloud-cpp/blob/7d81e4ead9f66cbb88d684397b757fe13b661b34/ci/cloudbuild/builds/lib/integration.sh#L42

while the Kokoro builds run with the default for Spanner.

**Comment by devjgm** [comment]
> I think #5926 extended the integration test time for these tests, but in the GCB builds we override that:
> 
> https://github.com/googleapis/google-cloud-cpp/blob/7d81e4ead9f66cbb88d684397b757fe13b661b34/ci/cloudbuild/builds/lib/integration.sh#L42
> 
> while the Kokoro builds run with the default for Spanner.

Thanks.

I can bump the spanner test timeout in the GCB build, but we currently have a 3600s timeout on the whole GCB pipeline. 

https://github.com/googleapis/google-cloud-cpp/blob/7d81e4ead9f66cbb88d684397b757fe13b661b34/ci/cloudbuild/cloudbuild.yaml#L45

And that may not be enough.

I'm seeing if I can make that a parameter so that it's only bumped up for the `integration-daily` test and not the others.

**Comment by devjgm** [comment]
> > I think #5926 extended the integration test time for these tests, but in the GCB builds we override that:
> > https://github.com/googleapis/google-cloud-cpp/blob/7d81e4ead9f66cbb88d684397b757fe13b661b34/ci/cloudbuild/builds/lib/integration.sh#L42
> > 
> > while the Kokoro builds run with the default for Spanner.
> 
> Thanks.
> 
> I can bump the spanner test timeout in the GCB build, but we currently have a 3600s timeout on the whole GCB pipeline.
> 
> https://github.com/googleapis/google-cloud-cpp/blob/7d81e4ead9f66cbb88d684397b757fe13b661b34/ci/cloudbuild/cloudbuild.yaml#L45
> 
> And that may not be enough.
> 
> I'm seeing if I can make that a parameter so that it's only bumped up for the `integration-daily` test and not the others.

Actually, as long as the tests run in parallel, (they should), 3600s is probably enough overall. I'll leave that for now and see if we hit it.

File: ci/cloudbuild/builds/lib/integration.sh [hunk 2]
```diff
@@ -153,7 +153,7 @@ function integration::bazel_with_emulators() {
   # - GOOGLE_CLOUD_CPP_SPANNER_DEFAULT_ENDPOINT
   io::log_h2 "Running Spanner integration tests"
   bazel "${verb}" "${args[@]}" --test_tag_filters=integration-test \
-    google/cloud/spanner/...
+    --test_timeout=3600 google/cloud/spanner/...
```
Comment (devbww) [inline ci/cloudbuild/builds/lib/integration.sh:156]: Why do we need that?  Do the BUILD-file values not apply?

Comment (devjgm) [reply ci/cloudbuild/builds/lib/integration.sh:156]: It's set to 600 above, so I need to override it here.

Also, `eternal` implies a timeout of 3600s, so this is setting it to the equivalent of eternal.

https://docs.bazel.build/versions/master/test-encyclopedia.html#role-of-the-test-runner

Comment (devjgm) [reply ci/cloudbuild/builds/lib/integration.sh:156]: ... of course, I could remove the `--test_timeoout` above and always rely on the per-test size setting... that may be better.

**Comment by devjgm** [comment]
FYI, I got a successful run: https://pantheon.corp.google.com/cloud-build/builds/9293a306-94bf-49e6-b978-5f8836ef43f4?project=cloud-cpp-testing-resources ~35 minutes

**Pull request merged**